### PR TITLE
Plans: Prevent persistent loading state on `Plans`

### PR DESCRIPTION
**Pull request by drewblaisdell**
Previously, if `sites-list` was empty on an initial load, it would not trigger a render() on `Plans` when the sites eventually loaded. This is because `Plans` used its `selectedSite` prop to determine which site ID to request plans for, which is problematic:

- `selectedSite` is false when no site is selected or sites aren't loaded.
- When sites do load, `getSelectedSite()` isn't called in the controller, so `selectedSite` remains false until the given controller method is called again.

This PR:

- Removes `selectedSite` as a prop of `Plans`, in favor of using `sites.getSelectedSite()` inside that component.
- Updates `props.fetchSitePlans` to properly return the initial state of a site's plan before they are loaded and only request site plans when a selected site is present.
- Updates `Checkout` to actually clear the site plans when checking out with a free trial. We had code that was supposed to do this here, but it created a thunk without passing it to the dispatcher.

**Testing**
- Call `localStorage.clear()` in your console.
- Visit http://calypso.localhost:3000/plans/:site for any site and assert that the plans or trial hub loads. We should try this for sites with the free plan, sites with a paid plan, and sites with a free trial.

- [x] Product review
- [x] Code review

### CODE REVIEW

**Comment by fabianapsimoes** [comment]
Product :+1:

File: client/my-sites/plans/main.jsx [hunk 1]
```diff
@@ -76,7 +75,7 @@ var Plans = React.createClass( {
 	},
 
 	redirectToDefault() {
-		page.redirect( paths.plans( this.props.selectedSite.slug ) );
+		page.redirect( paths.plans( this.props.getSelectedSite().slug ) );
```
Comment (scruffian) [inline client/my-sites/plans/main.jsx]: Shouldn't this be `this.props.sites.getSelectedSite()`?

**Comment by scruffian** [comment]
I fixed one thing.

File: client/my-sites/plans/main.jsx (hunk 1 above)
Comment (drewblaisdell) [inline client/my-sites/plans/main.jsx]: Thanks for catching that.

**Pull request merged**